src/cswinrt/code_writers.h

```diff
@@ -2467,14 +2544,36 @@ db_path.stem().string());
         );
     }
 
+    void write_event_source_generic_args(writer& w, cswinrt::type_semantics eventTypeSemantics);
+
     void write_event_source_ctor(writer& w, Event const& evt)
     {
+        bool isSpecialCase = false;
```

Comment:
note that for_typedef can return a value, which you could use to short-circuit return (and eliminate isSpecialCase)